.github/scripts/lint_test_ownership.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+'''
+Test ownership was introduced in https://github.com/pytorch/pytorch/issues/66232.
+
+This lint verifies that every Python test file (file that matches test_*.py or *_test.py in the test folder)
+has valid ownership information in a comment header. Valid means:
+  - The format of the header follows the pattern "# Owner(s): ["list", "of owner", "labels"]
+  - Each owner label actually exists in PyTorch
+  - Each owner label starts with "module: " or "oncall: " or is in ACCEPTABLE_OWNER_LABELS
+
+This file is expected to run in the root directory of pytorch/pytorch.
+'''
+
+import fnmatch
+import json
+import subprocess
+import sys
+from pathlib import Path
+from typing import List
+
+
+# Team/owner labels usually start with "module: " or "oncall: ", but the following are acceptable exceptions
+ACCEPTABLE_OWNER_LABELS = ["NNC", "high priority"]
+GLOB_EXCEPTIONS = [
+    "test/run_test.py"
+]
+
+PYTORCH_ROOT = Path(__file__).resolve().parent.parent.parent
+TEST_DIR = PYTORCH_ROOT / "test"
+CURRENT_FILE_NAME = Path(__file__).resolve().relative_to(PYTORCH_ROOT)
+
+
+def get_all_test_files() -> List[Path]:
+    test_files = list(TEST_DIR.glob("**/test_*.py"))
+    test_files.extend(list(TEST_DIR.glob("**/*_test.py")))
+    return [f for f in test_files if any([fnmatch.fnmatch(f, g) for g in GLOB_EXCEPTIONS])]
+
+
+def get_pytorch_labels() -> List[str]:
```

Comment:
This function makes 11 API calls. 1 to get the header, and 10 for each page of labels. When testing, I ran into API limit but will this be a problem in CI?

Reply:
I can lessen it to use maybe 4-5 API calls, but would this matter?

Reply:
Could we hardcode labels instead? They shouldn’t change too often and that avoids the general network badness we’ve seen

Reply:
I'm less keen to that because I think one of the points of this lint is to make sure the label is actually in pytorch/pytorch and manually maintaining a list seems either too extra (like if we have another bot to update that list every once in a while) or somewhat defeats the purpose.

Reply:
Couldn't we just grab these through the API and cache them somewhere on S3?

And then we could have a github action that updates the s3 cache every single time there's a new label created / deleted?

Reply:
The S3 list is fine, but do you anticipate the current approach being problematic on CI? Would it be because of the network reliance issue or the fact that CI machines may hit the limit?

Reply:
If this lint is run on every push for every PR and on master then yes we'll definitely hit the limit, especially considering that this is an unauthenticated API call